src/generala/utils/CombinationFinder.java

```diff
@@ -0,0 +1,240 @@
+    public Map<CombinationEnum, Integer> findCombinationsInPlayerDiceRoll(Player player) {
+        Map<CombinationEnum, Integer> combinationTreeMapReversed = new TreeMap<>(Collections.reverseOrder());
+        for (Map.Entry<Integer, Integer> currentSideEntrySet : player.getDiceRoll()
+                .getEachSideDuplicatesTreeMapReversed()
+                .entrySet()) {
+            int currentSideKey = currentSideEntrySet.getKey();
+            int currentSideValue = currentSideEntrySet.getValue();
+            if (hasGenerala(currentSideValue)) {
+                addGenerala(currentSideKey, combinationTreeMapReversed);
+                return combinationTreeMapReversed;
+            }
+            if (canBreakEarly(player.getRolledCombinations().size(), combinationTreeMapReversed.size())) {
+                break;
+            }
+            if (currentSideValue >= TWO_DUPLICATES) {
+                findNonCompoundCombinations(currentSideKey, currentSideValue);
+            }
+        }
+        callAddMethods(player, combinationTreeMapReversed);
+        return combinationTreeMapReversed;
+    }
+
+    //HELPER METHODS
+    private void findNonCompoundCombinations(int currentSide, int currentSideDuplicates) {
+        if (currentSideDuplicates >= FOUR_OF_A_KIND_SIZE) {
+            if (!hasBiggestFourOfAKind()) {
+                setBiggestFourOfAKind(currentSide);
+            }
+        }
+        if (currentSideDuplicates >= TRIPLE_SIZE) {
+            if (!hasBiggestTriple()) {
+                setBiggestTriple(currentSide);
+            }
+        }
+        if (currentSideDuplicates >= PAIR_SIZE) {
+            if (!hasBiggestPair()) {
+                setBiggestPair(currentSide);
+            } else if (!hasSecondBiggestPair()) {
+                setSecondBiggestPair(currentSide);
+            }
+        }
+    }
+
+    private void callAddMethods(Player player, Map<CombinationEnum, Integer> combinationTreeMap) {
+        EnumSet<CombinationEnum> playerRolledCombinations = player.getRolledCombinations();
+        if (hasBiggestFourOfAKind()) {
+            addFourOfAKind(playerRolledCombinations, combinationTreeMap);
+        }
+        if (hasBiggestTriple()) {
+            addTriple(playerRolledCombinations, combinationTreeMap);
+        }
+        if (hasBiggestPair()) {
+            addPair(playerRolledCombinations, combinationTreeMap);
+        }
+        if (hasSecondBiggestPair()) {
+            addDoublePair(playerRolledCombinations, combinationTreeMap);
+        }
+        if (hasBiggestTriple() && hasBiggestPair()) {
+            addFullHouse(playerRolledCombinations, combinationTreeMap);
+        }
+        if (hasStraight(player.getDiceRoll().getEachSideDuplicatesTreeMapReversed().keySet())) {
+            addStraight(playerRolledCombinations, combinationTreeMap);
+        }
+        resetInstanceVariablesForCombinationCreating();
+    }
+
+    private void resetInstanceVariablesForCombinationCreating() {
```

Comment:
Perhaps there is a better name for this method. `resetCombinationVariables` you do not need to put anything java specific in method names. The person who reads the code is aware that these are instance variables.